File: sentimental-cash/cash.py
from cs50 import get_float
from math import floor

while True:
    c = get_float("Change Owed: ")  # Use `get_float` for decimal input
    if c > 0:  # Adjusted condition to check for positive values
        break

# Convert dollars to cents to work with integers
c = round(c * 100)

# Initialize variables for coins
quarters = 0
dimes = 0
nickels = 0

if c / 25 >= 1:
    quarters = c // 25

remaining_cents = c % 25

if remaining_cents >= 10:
    dimes = remaining_cents // 10

remaining_cents %= 10

if remaining_cents >= 5:
    nickels = remaining_cents // 5

remaining_cents %= 5

tot_coins = quarters + dimes + nickels + remaining_cents
print(f"{tot_coins}")

File: cash/cash.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

int main(void)
{
    int c;
    int quarters = 0;
    int nickels = 0;
    int dimes = 0;
    int remaining_cents = 0;

    do
    {
        c = get_int("Change Owed: ");
    }
    while (c <= 0);

    if (c / 25 >= 1)
    {
        quarters = floor(c / 25);
    }

    remaining_cents = c % 25;

    if (remaining_cents >= 10)
    {

        dimes = floor(remaining_cents / 10);
    }

    remaining_cents = remaining_cents % 10;

    if (remaining_cents >= 5)
    {

        nickels = floor(remaining_cents / 5);
    }

    remaining_cents = c % 5;

    int tot_coins = quarters + dimes + nickels + remaining_cents;
    printf("%d\n", tot_coins);
}
